components/selection/selection-observer-mixin.js

```diff
@@ -16,21 +16,21 @@ export const SelectionObserverMixin = superclass => class extends superclass {
 			 * @type {object}
 			 */
 			selectionInfo: { type: Object },
-			_provider: { type: Object, attribute: false }
+			_provider: { state: true }
 		};
 	}
 
 	constructor() {
 		super();
 		this.selectionInfo = new SelectionInfo();
-		this._provider = null;
+		this.__provider = null;
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
 		// delay subscription otherwise import/upgrade order can cause selection mixin to miss event
 		requestAnimationFrame(() => {
-			if (this.selectionFor) return;
+			if (this.selectionFor || this._provider) return;
 
 			const evt = new CustomEvent('d2l-selection-observer-subscribe', {
```

Comment:
With the change to a setter, there's a _slight_ change to the logic here:

- when this event resolves and sets `this._provider = evt.detail.provider`, it'll go through the setter logic
- that will trigger an extra `newProvider.subscribeObserver(this)` that wasn't previously being triggered - it should have no effect, but is a bit of extra computation (re-triggers `_updateSelectionObservers`)
- an alternative here would be to set `this.__provider` (the private var) and then manually trigger `this.requestUpdate('_provider', oldProvider)`